include/lazy_importer.hpp

```diff
@@ -509,7 +507,7 @@ namespace li { namespace detail {
             if(!cached)
                 cached = Derived::template get<void*, Enum>();
 
-            return (T)(cached);
+            return reinterpret_cast<T>(cached);
```

Comment:
The change from C style to reinterpret cast for a generic type is a dangerous one. They were C style casts for a reason and I'd like to keep them that way.

Reply:
I get it. I'm going to compare the assembly generation of the two styles before any further comment, but for now I'll revert to old style casts.

Reply:
It's not about the assembly generation. It's just that C style casts are not equivalent to reinterpret_cast semantically, so this change _may_ be breaking someones code.

Reply:
I see. Then it's better to temporarily disable the related warning flag before the inclusion of the header. Thanks.